collection-scripts/gather_sos

```diff
@@ -153,23 +159,48 @@ dest_svc_path () {
 
 mkdir -p "${SOS_PATH_NODES}"
 
+nodes=''
+
+# SELECT NODES BY REQUESTED SERVICE
 # Get list of nodes and service label for each of the OpenStack service pods
 # Not using -o jsonpath='{.spec.nodeName}' because it uses space separator
-svc_nodes=$(/usr/bin/oc -n openstack get pod -l service --no-headers -o=custom-columns=NODE:.spec.nodeName,SVC:.metadata.labels.service,NAME:.metadata.name)
-nodes=''
-while read -r node svc name; do
-    svc_path=$(dest_svc_path "$svc")
-    if [[ -n "$svc_path" ]]; then
+# shellcheck disable=SC2128 # it's enough to check the first element to see if it's empty
+if [[ -n "${SOS_SERVICES}" ]]; then
+    svc_nodes=$(/usr/bin/oc -n openstack get pod -l service --no-headers -o=custom-columns=NODE:.spec.nodeName,SVC:.metadata.labels.service,NAME:.metadata.name)
```

Comment:
this assumes that services always get install into the openstack namespace. how about when they got installed in a different namespace?

Reply:
That assumption is not new to the PR.

In any case, as discussed earlier I will rework the approach to adapt it to:

- 3 master/worker nodes
- 3 master + 3 worker nodes
- Logs for pods in other namespaces